docs/Localization.md

```diff
@@ -0,0 +1,80 @@
+# Template localization 
+
+The templates are localizable. If the template is localized for the language matching the current locale, its elements will appear in that language in UX including `dotnet new` and Visual Studio.
+Localizable elements are:
+- name 
+- author
+- description
+- symbols
+  - description
+  - display name
+  - description and display name for choices for  choice parameters
+- post actions
+  - description
+  - manual instructions
+
+Localization files should be put to `localize` folder inside `.template.config`. Format of localization file is JSON, there is one file per language.
+Naming convention for localization files is: `templatestrings.<lang code>.json`. Language code should match name of [cultures](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.name?view=net-6.0#system-globalization-cultureinfo-name).
+Content of JSON are key-value pairs, where:
+- key is the reference to element of `template.json` to be localized. if the element is child, the full path using `/` delimiter should be given, example: `symbols/Framework/choices/netstandard2.1/description`.
+- value is localization of the element given in the key. 
+
+Examples of localization files can be found [here](https://github.com/dotnet/templating/tree/main/template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/localize).
+
+These files can be parsed by template engine when loading information about the template. Template engine API returns the information for localizable properties in language matching the current UI culture (if localization is available).
+
+### Post action localization
+Ensure to add `id` property for all post actions in `template.json` prior setting up the localization. `id` should be unique within the template. 
```

Comment:
```suggestion
Ensure that all post actions in `template.json` have an `id` property prior setting up the localization. The `id` should be unique within the template. 
```

It might be a nice thing to add to validate this in the MSBuild Tasks. I might look at adding this :)

Reply:
It is already validated in MSBuild task.

Reply:
Ah, I see - it happens via the ` LocalizationKeyIsNotUniqueException` during the template string extraction process.